lib/exqlite/sqlite3.ex

```diff
@@ -78,6 +78,27 @@ defmodule Exqlite.Sqlite3 do
   @spec step(db(), statement()) :: :done | :busy | {:row, []}
   def step(conn, statement), do: Sqlite3NIF.step(conn, statement)
 
+  @spec multi_step(db(), statement()) :: :busy | {:rows, [[]]} | {:done, [[]]}
+  def multi_step(conn, statement), do: multi_step(conn, statement, 50)
+
+  @spec multi_step(db(), statement(), integer()) ::
+          :busy | {:rows, [[]]} | {:done, [[]]}
+  def multi_step(conn, statement, chunk_size) do
+    case Sqlite3NIF.multi_step(conn, statement, chunk_size) do
+      :busy ->
+        {:error, "Database busy"}
+
+      {:error, reason} ->
+        {:error, reason}
+
+      {:rows, rows} ->
+        {:rows, Enum.reverse(rows)}
```

Comment:
can you leave a comment as to why we need the call to `Enum.reverse` here? I'm not entirely sure why it is needed (though it looks like it is indeed, by virtue of the test passing)

Reply:
guessing its a NIF thing

Reply:
Because of the nature of the linked lists that are in erlang, it's easier to just keep prepending in the nif. The other solution would be to allocate the chunk size in the nif and set the values there. Then when it is done, we call `enif_make_list_from_array`.

Pros and cons to both ways. Allocating the array and then calling `enif_make_list_from_array` requires us to manually free the underlying array before returning. Not a huge deal.